File: app/src/main/java/com/example/spacetraders/data/viewmodels/NewCharacterViewModel.java
package com.example.spacetraders.data.viewmodels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.support.annotation.NonNull;

import com.example.spacetraders.data.entity.GameDifficulty;
import com.example.spacetraders.data.entity.Universe;
import com.example.spacetraders.data.models.CharacterInteractor;
import com.example.spacetraders.data.models.Interactor;
import com.example.spacetraders.data.models.Model;
import com.example.spacetraders.data.entity.Character;
import com.example.spacetraders.data.models.ShipInteractor;
import com.example.spacetraders.data.models.UniverseInteractor;

public class NewCharacterViewModel extends AndroidViewModel {
    private CharacterInteractor characterInteractor;
    private UniverseInteractor universeInteractor;


    /**
     * Creates ViewModel that will deal with info from the new character activity
     * @param application no clue what this means
     */
    public NewCharacterViewModel(@NonNull Application application) {
        super(application);
        characterInteractor = Model.getInstance().getCharacterInteractor();
        universeInteractor = Model.getInstance().getUniverseInteractor();
    }

    /**
     * creates the character and stores it into the Model Interactor
     * @param character the character data that is the new character
     */
    public void createCharacter(Character character) {characterInteractor.createCharacter(character);}

    public Character getCharacter() {return characterInteractor.getCharacter();}

    /**
     * Gets the Universe and stores it to the Universe Interactor
     * @param universe the Universe that was randomly generated at character creation
     */
    public void createUniverse(Universe universe) { universeInteractor.createUniverse(universe);}

    public Universe getUniverse() {return universeInteractor.getUniverse();}
}

File: app/src/main/java/com/example/spacetraders/data/viewmodels/NewCharacterViewModel.kt
package com.example.spacetraders.data.viewmodels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel

import com.example.spacetraders.data.entity.GameDifficulty
import com.example.spacetraders.data.entity.Universe
import com.example.spacetraders.data.models.CharacterInteractor
import com.example.spacetraders.data.models.Interactor
import com.example.spacetraders.data.models.Model
import com.example.spacetraders.data.entity.Character
import com.example.spacetraders.data.models.ShipInteractor
import com.example.spacetraders.data.models.UniverseInteractor

class NewCharacterViewModel
/**
 * Creates ViewModel that will deal with info from the new character activity
 * @param application no clue what this means
 */
(application: Application) : AndroidViewModel(application) {
    private val characterInteractor: CharacterInteractor
    private val universeInteractor: UniverseInteractor

    val character: Character?
        get() = characterInteractor.character

    val universe: Universe?
        get() = universeInteractor.universe


    init {
        characterInteractor = Model.instance.characterInteractor
        universeInteractor = Model.instance.universeInteractor
    }

    /**
     * creates the character and stores it into the Model Interactor
     * @param character the character data that is the new character
     */
    fun createCharacter(character: Character) {
        characterInteractor.createCharacter(character)
    }

    fun setName(name: String) {
        characterInteractor.setName(name)
    }

    fun setDifficulty(difficulty: GameDifficulty) {
        characterInteractor.setDifficulty(difficulty)
    }

    /**
     * Gets the Universe and stores it to the Universe Interactor
     * @param universe the Universe that was randomly generated at character creation
     */
    fun createUniverse(universe: Universe) {
        universeInteractor.createUniverse(universe)
    }
}
